cas-server-support-openid/src/main/java/org/jasig/cas/support/openid/authentication/principal/OpenIdService.java

```diff
@@ -1,219 +1,234 @@
+        this.openIdPrefixUrl = openIdPrefixUrl;
+    }
+
+    /**
+     * Generates an Openid response.
+     * If no ticketId is found, response is negative.
+     * If we have a ticket id, then we check if we have an association.
+     * If so, we ask OpenId server manager to generate the answer according with the existing association.
+     * If not, we send back an answer with the ticket id as association handle.
+     * This will force the consumer to ask a verification, which will validate the service ticket.
+     * @param ticketId the service ticket to provide to the service.
+     * @return the generated authentication answer
+     */
+    @Override
+    public Response getResponse(final String ticketId) {
+        final Map<String, String> parameters = new HashMap<>();
+        if (ticketId != null) {
+
+            final ServerManager manager = (ServerManager) ApplicationContextProvider.getApplicationContext().getBean("serverManager");
+            final CentralAuthenticationService cas = ApplicationContextProvider.getApplicationContext()
+                                                .getBean("centralAuthenticationService", CentralAuthenticationService.class);
+            boolean associated = false;
+            boolean associationValid = true;
+            try {
+                final AuthRequest authReq = AuthRequest.createAuthRequest(requestParameters, manager.getRealmVerifier());
+                final Map parameterMap = authReq.getParameterMap();
+                if (parameterMap != null && parameterMap.size() > 0) {
+                    final String assocHandle = (String) parameterMap.get("openid.assoc_handle");
+                    if (assocHandle != null) {
+                        final Association association = manager.getSharedAssociations().load(assocHandle);
+                        if (association != null) {
+                            associated = true;
+                            if (association.hasExpired()) {
+                                associationValid = false;
+                            }
+                        }
+
+                    }
+                }
+            } catch (final MessageException me) {
+                LOGGER.error("Message exception : {}", me.getMessage(), me);
+            }
+
+            boolean successFullAuthentication = true;
+            Assertion assertion = null;
+            try {
+                if (associated) {
+                    if (associationValid) {
+                        assertion = cas.validateServiceTicket(ticketId, this);
+                        LOGGER.info("Validated openid ticket");
+                    } else {
+                        successFullAuthentication = false;
+                    }
+                }
+            } catch (final TicketException te) {
+                LOGGER.error("Could not validate ticket : {}", te.getMessage(), te);
+                successFullAuthentication = false;
+            }
+
+            final String id;
```

Comment:
Here is the change: if the `openid.identifier` is to be defined, we use the principal found in the assertion.